### ISSUE TITLE
Email the user a message if there ticket gets rejected

### ISSUE THREAD

**Issue by webscreations** [osTicket/osTicket#2267]
Is there anyway the system can email the user a message if there ticket gets rejected.

I turned off "Accept email from unknown Users" as I only want registered users to use the system. I need it to send out a canned message or something so the sure knows they need to login to there account to open a new ticket.

I would appreciate some help with this

**Comment by webscreations** [comment]
Anyone got any ideas, thanks very much?

**Comment by greezybacon** [comment]
This is possible in v1.10. You can add a filter action to send a mail before the action to reject

**Comment by webscreations** [comment]
Hi would you mind helping me, what find of filter would I do to make this work. Thanks

**Comment by greezybacon** [comment]
<img width="948" alt="screen shot 2015-07-24 at 5 11 15 pm" src="https://cloud.githubusercontent.com/assets/672074/8885623/121cfc76-3227-11e5-9762-03440fa2f2af.png">

**Comment by sriddick** [comment]
Hi Guys,

I have encountered the exact same issue as above. We turned off "Accept email from unknown Users" as we only want registered users to use the system due to a lot of spam tickets being logged. Users aleady registered can simply e-mail and a ticket is logged but non-registered users are not able to (due to the setting) 

The issue is that the sender may be a legitimate client and when sending an e-mail which is rejected by OSTicket, there is no bounce message that returns to say the mail delivery / ticket creation was unsuccessful. The solution provided above requires you to provide the senders domain in order for the rule to work, but obviously this will not solve the problem if it is a new client and they are simply making contact via e-mail for the first time. It would be far better if OSTicket could provide a canned reply or similar for rejected tickets based on the user not being registered. This way the message could let the user know that the ticket was not logged, as well as direct them to the URL to register. 

Is there any way of achieving this?

**Comment by ntozier** [comment]
You mean like @greezybacon posted above about that you replied to?

**Comment by sriddick** [comment]
Yes, I have the same problem as @greezybacon.

**Comment by sriddick** [comment]
My apologies, I have the same problem as @webscreations. The solution provided by @greezybacon will not solve the problem because you need to specify the originating domain name or similar in order for the rule to match. This is a problem. It is not possible to simply have a canned response for rejected tickets?

**Comment by sriddick** [comment]
Anyone?

**Comment by ntozier** [comment]
This is not a feature of osTicket at this time.  I opened a feature request for this back in June.
https://github.com/osTicket/osTicket/issues/3171

**Comment by SoulGit** [comment]
I realize this issue is dated, however is the solution provided by greezybacon still valid in 1.15.2?  If I select Send an Email in my filter as the sole option, I successfully receive an e-mail.  If I add the Reject Ticket following Send an Email, I do not receive a notification e-mail, though the ticket is rejected.  Thank you.

**Comment by JediKev** [comment]
@SoulGit 

Unfortunately no, not at the moment. You will need two separate filters: one to send an email and a second one to reject the ticket.

Cheers.

**Comment by SoulGit** [comment]
Thank you for the super quick response @JediKev!  This seems like the perfect solution so I gave it a whirl just now, unfortunately it doesn't seem to work. I now have two filters, exactly the same rule, does not match regex. The email rule is 10 with Send an Email as the sole option, the reject rule is 11 with Reject Ticket as the sole option.  I see the Ticket Denied in the dashboard, but no e-mail was sent.  Anything else I can try here? (Brief update after additional testing: if I disable the Reject rule, then I do receive the email, so it seems to be working very similarly to when the Filter Actions were combined in a single rule).

**Comment by SoulGit** [comment]
@JediKev 

I am still struggling with this issue.  I have completely deleted my filters and created two fresh items as described above.  Whenever there is a reject rule, the email isn't sent.  What further info can I provide to help diagnose / resolve?

Thanks!

Adam

**Comment by JediKev** [comment]
@SoulGit 

Yea, there's this bit of code:
- https://github.com/osTicket/osTicket/blob/develop/include/class.filter.php#L339-L342

With the above, the Send Email filter action will be skipped because it's not post-create yet and the Reject filter action will execute, thereby stopping the creation altogether.

TLDR; You cannot do this unless you modify the codebase.

Cheers.
